h5io/_h5io.py

```diff
@@ -423,3 +425,63 @@ def __init__(self):
 
     def __del__(self):
         rmtree(self._path, ignore_errors=True)
+
+
+def _list_file_contents(h5file):
+    if 'h5io' not in h5file.keys():
+        raise ValueError('h5file must contain h5io data')
+
+    # Set up useful variables for later
+    h5file = h5file['h5io']
+    root_title = h5file.attrs['TITLE']
+    n_space = np.max([(len(key), len(val.attrs['TITLE']))
+                      for key, val in h5file.items()]) + 2
+
+    # Create print strings
+    strs = ['Root type: {} | Items: {}\n'.format(root_title, len(h5file))]
+    for key, data in h5file.items():
+        type_str = data.attrs['TITLE']
+        str_format = '{{:<{}}}'.format(n_space)
+        if type_str == 'ndarray':
+            desc = 'Shape: {0}'
+            desc_val = data.shape
+        elif type_str == 'pd_dataframe':
+            desc = 'Shape: {0}'
+            desc_val = data['values'].shape
+        elif type_str in ('unicode', 'ascii', 'str'):
+            desc = 'Text: {0}'
+            decoder = 'utf-8' if type_str == 'unicode' else 'ASCII'
+            cast = text_type if type_str == 'unicode' else str
+            data = cast(np.array(data).tostring().decode(decoder))
+            desc_val = data[:10] + '...' if len(data) > 10 else data
+        else:
+            desc = 'Items: {0}'
+            desc_val = len(data)
+        this_str = ('{{}} Key: {0} | Type: {0} | ' + desc).format(str_format)
+        this_str = this_str.format(tab_str, key, type_str, desc_val)
+        strs.append(this_str)
+    out_str = '\n'.join(strs)
+    print(out_str)
+
+
+def list_file_contents(h5file):
+    """List the contents of an h5io file.
+
+    This will list the root and one-level-deep contents of the file.
+
+    Parameters
+    ----------
+    h5file : str
+        The path to an h5io hdf5 file.
+    """
+    h5py = _check_h5py()
+    err = 'h5file must be an h5py File object, not {}'
+    if isinstance(h5file, str):
+        with h5py.File(h5file, 'r') as f:
+            if not isinstance(f, h5py.File):
+                raise ValueError(err.format(type(h5file)))
```

Comment:
oops good catch, I'm trying to debug some python3 string errors and will fix that in the next push too